pkg/auth/oauth/registry/tokentimeouts.go

```diff
@@ -186,15 +196,17 @@ func (a *oauthTokenTimeoutValidator) flush(flushHorizon time.Time) {
 
 		td := a.data[tdr.name]
 		delta := a.clientTimeout(td.token.ClientName)
-		var newtimeout int32
-		if delta > 0 {
```

Comment:
somehow you lost here a previous } else { newtimeout = 0 } branch.
this means that if the oauthclient has been changed so that tokens do not have a timeout anymore (AccessTokenTimeoutSeconds == 0), what will happen instead is that the token will see it's timeout not updated at all as delta is 0.

Reply:
Fixed.